source/ftrack_connect_pipeline_3dsmax/asset/dcc_object.py

```diff
@@ -0,0 +1,156 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2014-2022 ftrack
+
+import logging
+
+from ftrack_connect_pipeline.asset.dcc_object import DccObject
+from ftrack_connect_pipeline_3dsmax.constants import asset as asset_const
+from ftrack_connect_pipeline_3dsmax.utils import custom_commands as max_utils
+
+# import maya.cmds as cmds
+
+
+class MaxDccObject(DccObject):
+    '''MaxDccObject class.'''
+
+    ftrack_plugin_id = asset_const.FTRACK_PLUGIN_ID
+    '''Plugin id used on some DCC applications '''
+
+    def __init__(self, name=None, from_id=None, **kwargs):
+        '''
+        If the *from_id* is provided find an object in the dcc with the given
+        *from_id* as assset_info_id.
+        If a *name* is provided create a new object in the dcc.
+        '''
+        self.logger = logging.getLogger(
+            '{0}.{1}'.format(__name__, self.__class__.__name__)
+        )
+        super(MaxDccObject, self).__init__(name, from_id, **kwargs)
+
+    def __setitem__(self, k, v):
+        '''
+        Sets the given *v* into the given *k* and automatically set the
+        attributes of the current self :obj:`name` on the DCC
+        '''
+        # cmds.setAttr('{}.{}'.format(self.name, k), l=False)
+        # if k == asset_const.VERSION_NUMBER:
+        #     cmds.setAttr('{}.{}'.format(self.name, k), v, l=True)
+        # elif k == asset_const.REFERENCE_OBJECT:
+        #     cmds.setAttr(
+        #         '{}.{}'.format(self.name, k),
+        #         str(self.name),
+        #         type="string",
+        #         l=True,
+        #     )
+        # elif k == asset_const.IS_LATEST_VERSION:
+        #     cmds.setAttr('{}.{}'.format(self.name, k), bool(v), l=True)
+        #
+        # elif k == asset_const.DEPENDENCY_IDS:
+        #     cmds.setAttr(
+        #         '{}.{}'.format(self.name, k),
+        #         *([len(v)] + v),
+        #         type="stringArray",
+        #         l=True
+        #     )
+        #
+        # else:
+        #     cmds.setAttr(
+        #         '{}.{}'.format(self.name, k), v, type="string", l=True
+        #     )
+        super(MaxDccObject, self).__setitem__(k, v)
+
+    def create(self, name):
+        '''
+        Creates a new dcc_object with the given *name* if doesn't exists.
+        '''
+        if self._name_exists(name):
+            error_message = "{} already exists in the scene".format(name)
+            self.logger.error(error_message)
+            raise RuntimeError(error_message)
+
+        # dcc_object = cmds.createNode(asset_const.FTRACK_PLUGIN_TYPE, name=name)
```

Comment:
Remove unused code, But instead you have to create a group if I remember correctly.

Reply:
DCC object will be attended in a future PR